File: 2015/day20/p02.py
#!/usr/bin/env python3
# Copyright (c) 2025 Adam Mullins
#
# Solution for Advent of Code 2015, Day 02, Part 2
#   https://adventofcode.com/2015/day/2#part2
from array import array

n = 29000000
houses = array("L", (0 for _ in range(n + 1)))

for i in range(1, n + 1):
    count = 0
    j = i
    while count <= 50 and j < n + 1:
        houses[j] += i * 11
        count += 1
        j += i

for house, presents in enumerate(houses):
    if presents >= n:
        print(f"{house} got {presents} presents.")
        break

File: 2016/day02/p02.mjs
#!/usr/bin/env node
// Copyright (c) 2025 Adam Mullins
//
// Solution for Advent of Code 2016, Day 02, Part 2
//   https://adventofcode.com/2016/day/2#part2
"use strict";
import fs from "node:fs";


const keypad = ["       ",
                "   1   ",
                "  234  ",
                " 56789 ",
                "  ABC  ",
                "   D   ",
                "       "];
let row = 3;
let col = 1;
let code = "";

const input = fs.
      readFileSync(0, "ascii").
      trim().
      split("\n");

for(const line of input) {
    for(const c of line) {
        let tmp_row = row,
            tmp_col = col;
        if(c === "L") {--tmp_col;}
        if(c === "R") {++tmp_col;}
        if(c === "U") {--tmp_row;}
        if(c === "D") {++tmp_row;}
        if(keypad[tmp_row][tmp_col] !== " ") {
            row = tmp_row;
            col = tmp_col;
        }
    }
    code += keypad[row][col];
}

console.log(code);
